support for the `vnet_integration_enabled` and `subnet_id` properties (#30559)
hashicorp/terraform-provider-azurerm: website/docs/r/kubernetes_cluster.html.markdown

* `authorized_ip_ranges` - (Optional) Set of authorized IP ranges to allow access to API server, e.g. ["198.51.100.0/24"].

* `subnet_id` - (Optional) The ID of the Subnet where the API server endpoint is delegated to.

> * `virtual_network_integration_enabled` - (Optional) Should API Server VNet Integration be enabled? Defaults to `false`.

Reviewer: Could you update this to a statement rather than a question? (old pattern, new props should ideally be something like my suggestion below)

```suggestion
* `virtual_network_integration_enabled` - (Optional) Whether to enable virtual network integration for the API Server. Defaults to `false`.
```
